### ISSUE TITLE
資訊

### ISSUE THREAD

**Issue by 11330148** [11330148/--1#1]
import pandas as pd
from sklearn.ensemble import RandomForestRegressor
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import LabelEncoder
import gradio as gr

# 假設你已經整理好的資料存在 data.csv（需包含價格在最後一欄）
# 這裡簡單示意模擬資料（你可以替換成你自己的 CSV 資料）

data = pd.DataFrame({
    '區域': ['信義區', '松山區', '大安區', '信義區', '松山區'],
    '建物坪數': [30.5, 20.1, 40.0, 25.2, 18.7],
    '屋齡': [10, 5, 20, 8, 3],
    '樓層': [7, 3, 10, 5, 2],
    '車位': ['塔式車位', '坡道平面', '無', '坡道機械', '升降機械'],
    '經度': [121.56, 121.55, 121.54, 121.57, 121.52],
    '緯度': [25.03, 25.05, 25.04, 25.02, 25.01],
    '價格': [2300, 1500, 2800, 2000, 1800]  # 單位：萬
})

# 對類別欄位進行編碼
le_area = LabelEncoder()
le_parking = LabelEncoder()
data['區域'] = le_area.fit_transform(data['區域'])
data['車位'] = le_parking.fit_transform(data['車位'])

# 特徵與目標
X = data.drop('價格', axis=1)
y = data['價格']

# 訓練模型
model = RandomForestRegressor(n_estimators=100, random_state=42)
model.fit(X, y)

# 預測函式
def predict_price(area, size, age, floor, parking, lng, lat):
    area_encoded = le_area.transform([area])[0]
    parking_encoded = le_parking.transform([parking])[0]
    features = [[area_encoded, size, age, floor, parking_encoded, lng, lat]]
    price = model.predict(features)[0]
    return f"預測價格：約 {round(price, 2)} 萬元"

# Gradio 介面
with gr.Blocks() as demo:
    gr.Markdown("## 🏙️ 台北市房價預測系統")

    area = gr.Dropdown(choices=le_area.classes_.tolist(), label="區域")
    size = gr.Number(label="建物坪數")
    age = gr.Number(label="屋齡（年）")
    floor = gr.Number(label="樓層")
    parking = gr.Dropdown(choices=le_parking.classes_.tolist(), label="車位型態")
    lng = gr.Number(label="經度", value=121.55)
    lat = gr.Number(label="緯度", value=25.04)
    output = gr.Textbox(label="預測價格")

    btn = gr.Button("開始預測")
    btn.click(predict_price, inputs=[area, size, age, floor, parking, lng, lat], outputs=output)

demo.launch()
